features/filesystem/littlefs/littlefs/lfs.c

```diff
@@ -1697,11 +1693,15 @@ int lfs_stat(lfs_t *lfs, const char *path, struct lfs_info *info) {
         info->size = entry.d.u.file.size;
     }
 
-    err = lfs_bd_read(lfs, cwd.pair[0],
-            entry.off + 4+entry.d.elen+entry.d.alen,
-            info->name, entry.d.nlen);
-    if (err) {
-        return err;
+    if (lfs_paircmp(entry.d.u.dir, lfs->root) == 0) {
+        strcpy(info->name, "/");
+    } else {
+        err = lfs_bd_read(lfs, cwd.pair[0],
+                entry.off + 4+entry.d.elen+entry.d.alen,
+                info->name, entry.d.nlen);
+        if (err) {
```

Comment:
Any reason you don't `return(err)` here?

Reply:
This is a heavily reoccuring pattern:
``` c
int err = lfs_do_thing();
if (err) {
    return err;
}
```

Which is effectively a form of exception handling. It's better to keep these sort of patterns consistent, because it makes it easier to recognize in context.

I do like to chain returns when the control flow passes completely to another function, but since this use is inside an if statement, it makes it harder to follow. If we needed to add an epilogue to this function it wouldn't be clear if the `return err` was supposed to be exceptional behavior or should be changed to continue to the end of the function.

But it's a very minor thing, so I wouldn't bat an eye if someone committed code that did just `return err`.